pkg/maps/policymap/policymap.go

```diff
@@ -288,21 +289,29 @@ func (key *PolicyKey) ToNetwork() PolicyKey {
 
 // GetPrefixLen determines the length that should be set inside the Key so that
 // the lookup prefix is correct in the BPF map key. The specified 'prefixBits'
-// indicates the number of bits in the DestPort that must match to match the entry in
-// the BPF policy map.
+// indicates the number of bits in Nexthdr and DestPort that must match to match
+// the entry in the BPF policy map.
 func GetPrefixLen(prefixBits uint16) uint32 {
-	return getStaticPrefixBits() + uint32(prefixBits)
+	return GetStaticPrefixBits() + uint32(prefixBits)
 }
 
 // newKey returns a PolicyKey representing the specified parameters in network
 // byte-order.
-func newKey(id uint32, dport uint16, proto u8proto.U8proto, trafficDirection trafficdirection.TrafficDirection) PolicyKey {
-	prefixLen := uint16(16)
-	if dport == 0 {
-		prefixLen = 0
+func newKey(prefixLen uint32, id uint32, dport uint16, proto u8proto.U8proto, trafficDirection trafficdirection.TrafficDirection) PolicyKey {
+	if prefixLen < GetStaticPrefixBits() {
```

Comment:
There's something a little odd happening with the API here, since the caller seems to need to know the internal details of the policykey structure layout to make a decision about the length to use, where if it's less than this static group of fields it'll be ignored but if it's more than that static group of fields it'll be trusted directly (even if the value is crazy high, like `0xFFFFFFFF`).

I guess this is part of the debugging commit but it would make more sense to me if these policymap entries had a more locked-down "portPrefixLen" which *must* be in the range 0-16.

Reply:
And this is the part where I realize the field should be uninitialized in all of the current usage, therefore 0 and always match the first if condition here ;)

Reply:
(Oh and I assumed port is *always* specified but it looks like there's minimal wildcarding support for that here too).